From: Jeremy Bicha <jeremy.bicha@canonical.com>
Date: Wed, 23 Mar 2022 14:40:16 -0400
Subject: gdesktopappinfo: Recognize gnome-console as a terminal app

GNOME Console (installed as /usr/bin/kgx ) is the default
GNOME terminal app as of GNOME 42

Related to: https://gitlab.gnome.org/GNOME/glib/-/issues/338

But see https://gitlab.freedesktop.org/xdg/xdg-specs/-/merge_requests/46
for a more comprehensive way of handling issues around the default
terminal app
---
 gio/gdesktopappinfo.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gio/gdesktopappinfo.c b/gio/gdesktopappinfo.c
index c29c309..c244272 100644
--- a/gio/gdesktopappinfo.c
+++ b/gio/gdesktopappinfo.c
@@ -2676,6 +2676,8 @@ prepend_terminal_to_vector (int    *argc,
         }
       else
         {
+          if (check == NULL)
+            check = g_find_program_in_path ("kgx");
           if (check == NULL)
             check = g_find_program_in_path ("tilix");
           if (check == NULL)
